Lib/os.py

```diff
@@ -692,7 +692,7 @@ def __delitem__(self, key):
             del self._data[encodedkey]
         except KeyError:
             # raise KeyError with the original key value
-            raise KeyError(key) from None
+            raise KeyError("Environment variable '{}' was requested, but not found.".format(key)) from None
```

Comment:
I would leave this one alone.   End users aren't typically in the business of deleting environment variables.  Also, the "was requested" is more confusing than a straight *KeyError*.